File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260917-054908-40691/analysis/own-tests-project/src/main/java/customerimporter/OldZipCode.java
package customerimporter;

import java.util.function.Supplier;

public class OldZipCode extends ZipCode {

    private final Integer value;

    // instance creation

    static boolean canHandle(String aZipCode) {
        return Character.isDigit(aZipCode.charAt(0));
    }

    // error messages

    public static String invalidOldZipCodeErrorDescription() {
        return "Invalid old zipcode";
    }

    // initialization

    // e.g. 1636
    public OldZipCode(String aZipCode) {
        super(aZipCode);
        value = Integer.parseInt(aZipCode);
    }

    // accessing

    public Object value() {
        return value;
    }

    // testing

    public boolean isOld() {
        return true;
    }

    public boolean isNew() {
        return false;
    }

    public Object oldZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return value;
    }

    public Object newZipCodeIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    // validation

    protected boolean isValid(String aZipCode) {
        return Strings.isAllDigits(aZipCode)
            && Strings.isBetween(Long.parseLong(aZipCode), 1000, 9999);
    }

    protected String invalidZipCodeErrorDescription() {
        return invalidOldZipCodeErrorDescription();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-211617-6449/analysis/own-tests-project/src/customerimporter/old_zip_code.py
from customerimporter.zip_code import ZipCode


class OldZipCode(ZipCode):

    # type

    @classmethod
    def can_handle(cls, a_zip_code):
        return a_zip_code[:1].isdigit()

    # validation

    @classmethod
    def assert_valid(cls, a_zip_code):
        cls._assert_number_between(a_zip_code, cls.minimum_code(), cls.maximum_code())

    @classmethod
    def maximum_code(cls):
        return 9999

    @classmethod
    def minimum_code(cls):
        return 1000

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid old zipcode"

    # code

    def code(self):
        return int(self._value)

    def old_code_if_none(self, a_none_block):
        return self.code()

    # testing

    def is_old(self):
        return True
